src/components/Facility/AssetCreate.tsx

```diff
@@ -182,180 +194,137 @@ const AssetCreate = (props: AssetProps) => {
+      errors.support_email = "Please enter valid email id";
+    }
+
+    if (form.notes && !form.serviced_on) {
+      errors.serviced_on = "Last serviced on date is required with notes";
+    }
+
+    return errors;
   };
 
-  const handleSubmit = async (e: React.SyntheticEvent, addMore: boolean) => {
-    e.preventDefault();
-    const validated = validateForm();
-    if (validated) {
-      setIsLoading(true);
-      const data: any = {
-        name: name,
-        asset_type: AssetType.INTERNAL,
-        asset_class: asset_class || "",
-        description: description,
-        is_working: is_working,
-        not_working_reason: is_working === "true" ? "" : not_working_reason,
-        serial_number: serial_number,
-        location: location,
-        vendor_name: vendor_name,
-        support_name: support_name,
-        support_email: support_email,
-        support_phone: support_phone.startsWith("1800")
-          ? support_phone
-          : parsePhoneNumber(support_phone),
-        qr_code_id: qrCodeId !== "" ? qrCodeId : null,
-        manufacturer: manufacturer,
-        warranty_amc_end_of_validity: warranty_amc_end_of_validity
-          ? dateQueryString(warranty_amc_end_of_validity)
-          : null,
-      };
-
-      if (last_serviced_on) {
-        data["last_serviced_on"] = dateQueryString(last_serviced_on);
-        data["note"] = notes ?? "";
-      }
+  const handleSubmitAsync = async (form: AssetT, addMore: boolean) => {
+    setIsLoading(true);
+
+    const data: any = {
+      name: form.name,
+      asset_type: AssetType.INTERNAL,
+      asset_class: form.asset_class || "",
+      description: form.description,
+      is_working: form.is_working,
+      not_working_reason:
+        form.is_working === true ? "" : form.not_working_reason,
+      serial_number: form.serial_number,
+      location: form.location,
+      vendor_name: form.vendor_name,
+      support_name: form.support_name,
+      support_email: form.support_email,
+      support_phone: form.support_phone?.startsWith("1800")
+        ? form.support_phone
+        : parsePhoneNumber(String(form.support_phone)),
+      qr_code_id: form.qr_code_id !== "" ? form.qr_code_id : null,
+      manufacturer: form.manufacturer,
+      warranty_amc_end_of_validity: form.warranty_amc_end_of_validity
+        ? dateQueryString(form.warranty_amc_end_of_validity)
+        : null,
+    };
+
+    if (form.serviced_on) {
+      data["last_serviced_on"] = dateQueryString(form.serviced_on);
+      data["note"] = form.notes ?? "";
+    }
 
-      if (!assetId) {
+    try {
+      if (!form.id) {
         const { res } = await request(routes.createAsset, { body: data });
         if (res?.ok) {
           Notification.Success({ msg: "Asset created successfully" });
+          // Handle "Add More" logic if necessary
           if (addMore) {
-            resetFilters();
+            // resetFilters();
```

Comment:
#8956 adds an additionalButtons prop to Form Component, which will allow to keep the add more functionality. Take a look and write code (commented out) to work with it.

We can merge this PR once 8956 is merged.

Reply:
@Jacobjeevan i suggest that `additionalButtons` prop  in #8956 should take `buttonDisable` property also, for each  button, in that way Create & Add More button will be enabled depending on if we have `assetId` or not,
although i have made necesssary changes in `AssetCreate`.

Reply:
@Jacobjeevan 
i need to pass buttonDisabled as of now? is that what you're saying?
may be like this
additionalButtons={[
                  {
                    type: "submit",
                    label: t("create_add_more"),
                    id: "create-asset-add-more-button",
                    _**buttonDisabled : assetId ? false : true,**_
                  },
   and submit handler 
onSubmit={async (obj) => {
                  await handleSubmitAsync(obj, _**Boolean(assetId)**_);
                }}
takes care of rest

Reply:
> @Jacobjeevan i need to pass buttonDisabled as of now? is that what you're saying? may be like this additionalButtons={[ { type: "submit", label: t("create_add_more"), id: "create-asset-add-more-button", _**buttonDisabled : assetId ? true : false,**_ }, and existing submit handler onSubmit={async (obj) => { await handleSubmitAsync(obj, _**Boolean(assetId)**_); }} takes care of rest

No I meant, there is no need for additionalButtons to have anything if assetId is present (i.e. you are editing an existing asset).

Reply:
> > @Jacobjeevan i need to pass buttonDisabled as of now? is that what you're saying? may be like this additionalButtons={[ { type: "submit", label: t("create_add_more"), id: "create-asset-add-more-button", _**buttonDisabled : assetId ? true : false,**_ }, and existing submit handler onSubmit={async (obj) => { await handleSubmitAsync(obj, _**Boolean(assetId)**_); }} takes care of rest
> 
> No I meant, there is no need for additionalButtons to have anything if assetId is present (i.e. you are editing an existing asset).


@Jacobjeevan Ah! i see
this right 
![image](https://github.com/user-attachments/assets/6355130b-84e1-4814-92c9-1e6cd38ac5b6)
onSubmit={async (obj) => {
                  await handleSubmitAsync(obj, !Boolean(assetId));
                }}
                className="rounded bg-white p-6 transition-all sm:rounded-xl sm:p-12"
                noPadding
                validate={AssetFormValidator}
                submitLabel={assetId ? t("update") : t("create_asset")}
                additionalButtons={
                  !assetId
                    ? [
                        {
                          type: "submit",
                          label: t("create_add_more"),
                          id: "create-asset-add-more-button",
                        },
                      ]
                    : []
                }
            but still, the addMore will be true if we don't have assetId, so both create and Create add more will have same logic (i.e they'll redirect to same page after creating the new asset)

currently i am able to see create asset is working same as create add more, once we have 8956 merged create add more will work same.

since the logic for submit is same for both buttons in create page, and only navigation after submission is dependent on addMore property in the arg of
onSubmit={async (obj) => {
                  await handleSubmitAsync(obj, !Boolean(assetId));
                }}
 currently what we are passing is not a right determinant of addMore!

Reply:
one possible solution is this!

onSubmit={async (obj) => {
                  const clickedButtonId = (document.activeElement as HTMLButtonElement)?.id;
                  const addMore = clickedButtonId === "create-asset-add-more-button";
                  await handleSubmitAsync(obj, addMore);
                }}
this is also working for Create Asset, lets see if it works when 8596 merges.

Reply:
@rajku-dev That will work, but you don't need to handle the buttonId lookup either.

BedCapacity was doing something similar, so Form component had to be modified to support it.
https://github.com/ohcnetwork/care_fe/blob/20174b7531bacbc474e666bed6e7b21dd25277d2/src/components/Facility/BedCapacity.tsx#L159

https://github.com/ohcnetwork/care_fe/blob/20174b7531bacbc474e666bed6e7b21dd25277d2/src/components/Form/Form.tsx#L67 passes buttonId to the submit function, so you work based off that; i.e. change addMore param to be a check for specific the buttonId.

Reply:
@Jacobjeevan got it!
                onSubmit={async (obj, buttonId) => {
                  await handleSubmitAsync(obj, buttonId == "create-asset-add-more-button");
                }}
               this right..

Reply:
> @Jacobjeevan got it! onSubmit={async (obj, buttonId) => { await handleSubmitAsync(obj, buttonId == "create-asset-add-more-button"); }} this right..

You don't need to do that; form component would pass in the buttonId, just modify the addMore check accordingly inside the handleSubmit.

Reply:
@Jacobjeevan I think it is already handling 
try {
      if (!form.id) {
        const { res } = await request(routes.createAsset, { body: data });
        if (res?.ok) {
          Notification.Success({ msg: "Asset created successfully" });
          // Handle "Add More" logic 
          if (addMore) {
            resetFilters();
            const pageContainer = window.document.getElementById("pages");
            pageContainer?.scroll(0, 0);
          } else {
            goBack();
          }
        }
      } 
    Form comp will pass the id of button that is clicked, if it is the same as Create & Add More btn id i.e create-asset-add-more-button , addMore will be true, which is handled accordingly inside the handler.

Reply:
@rajku-dev Yes, but to clarify, I meant moving the buttonId check inside handleSubmit, rather than passing it in.

Reply:
> @rajku-dev Yes, but to clarify, I meant moving the buttonId check inside handleSubmit, rather than passing it in.

@Jacobjeevan done.